docs/lib/auth/fragments/android/signin_web_ui/20_platform_specific_setup.md

```diff
@@ -1,26 +1,56 @@
 ## Update AndroidManifest.xml
+
+<amplify-block-switcher>
+<amplify-block name="v1.17.8+">
+Add the following activity to your app's `AndroidManifest.xml` file, replacing `myapp` with
+your redirect URI prefix if necessary:
+
+```xml
+<activity
+    android:name="com.amplifyframework.auth.cognito.activities.HostedUIRedirectActivity">
+    <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="myapp" />
+    </intent-filter>
+</activity>
+```
+
+You may now skip the instructions below for adding a response handler to your activity.
+
+</amplify-block>
+<amplify-block name="v1.17.7 and below">
+**Note:** These versions have known issues with sign-out after signing in via web UI.
+Please update to the latest version and follow the updated instructions for best results.
+
 Add the following activity to your app's `AndroidManifest.xml` file, replacing `myapp` with
 whatever value you used for your redirect URI prefix:
 
-  ```xml
-  <activity
-      android:name="com.amazonaws.mobileconnectors.cognitoauth.activities.CustomTabsRedirectActivity">
-      <intent-filter>
-          <action android:name="android.intent.action.VIEW" />
+```xml
+<activity
+    android:name="com.amazonaws.mobileconnectors.cognitoauth.activities.CustomTabsRedirectActivity">
+    <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="myapp" />
+    </intent-filter>
+</activity>
+```
 
-          <category android:name="android.intent.category.DEFAULT" />
-          <category android:name="android.intent.category.BROWSABLE" />
+<amplify-callout>
+These instructions have been updated since version 1.2.0. If you set this up for a version of Amplify prior to 1.2.0, be sure to remove the `intent-filter` with `android:scheme` from your own activity as well as the `singleInstance` launch mode.
+</amplify-callout>
 
-          <data android:scheme="myapp" />
-      </intent-filter>
-  </activity>
-  ```
+</amplify-block>
+</amplify-block-switcher>
 
+### Add Response Handler
 <amplify-callout>
-  These instructions have been updated since version 1.2.0. If you set this up for a version of Amplify prior to 1.2.0, be sure to remove the `intent-filter` with `android:scheme` from your own activity as well as the `singleInstance` launch mode.
+If you are using a version of Amplify 1.17.8 or above and have already declared HostedUIRedirectActivity in your manifest file, then you may skip this step.
```

Comment:
```suggestion
If you are using a version of Amplify 1.17.8 or above and have already declared `HostedUIRedirectActivity` in your manifest file, then you may skip this step.
```

Reply:
the doc site doesn't actually correctly parse these markdowns in `<amplify-callout>` sections unfortunately.